pkg/storage/snapshot/restore.go

```diff
@@ -820,6 +798,49 @@ func (t *vmRestoreTarget) doesTargetVMExist() bool {
 	return t.vm != nil
 }
 
+func searchDV(name string, newTemplates []kubevirtv1.DataVolumeTemplateSpec) bool {
+	for _, ndv := range newTemplates {
+		if name == ndv.Name {
+			return true
+		}
+	}
+	return false
+}
+
+func (t *vmRestoreTarget) createOrUpdateVM(snapshotVM *snapshotv1.VirtualMachine, newTemplates []kubevirtv1.DataVolumeTemplateSpec, newVolumes []kubevirtv1.Volume) *kubevirtv1.VirtualMachine {
+	var newVM *kubevirtv1.VirtualMachine
+	if !t.doesTargetVMExist() {
+		newVM = &kubevirtv1.VirtualMachine{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:        t.vmRestore.Spec.Target.Name,
+				Namespace:   t.vmRestore.Namespace,
+				Labels:      snapshotVM.Labels,
+				Annotations: snapshotVM.Annotations,
+			},
+			Spec:   *snapshotVM.Spec.DeepCopy(),
+			Status: kubevirtv1.VirtualMachineStatus{},
+		}
+
+	} else {
+		newVM = t.vm.DeepCopy()
+		newVM.Spec = *snapshotVM.Spec.DeepCopy()
+	}
+
+	// update Running state in case snapshot was on online VM
+	if newVM.Spec.RunStrategy != nil {
+		runStrategyHalted := kubevirtv1.RunStrategyHalted
+		newVM.Spec.RunStrategy = &runStrategyHalted
+	} else if newVM.Spec.Running != nil {
+		running := false
+		newVM.Spec.Running = &running
+	}
```

Comment:
We could use here a switch case instead of a else if

Reply:
Although I thought using switch case won't really simplify it but looking at the rest of code base similar pattern is present, so right to do it.